components/carousel.js

```diff
@@ -104,10 +105,14 @@ function CarouselOverflow ({ originalSrc, rel }) {
 }
 
 export function CarouselProvider ({ children }) {
+  const pathname = usePathname()
+
   const media = useRef(new Map())
   const showModal = useShowModal()
 
   const showCarousel = useCallback(({ src }) => {
+    window.history.pushState(null, '', pathname)
```

Comment:
This isn't a [shallow push](https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing). On backwards navigation, this closes the carousel by reloading the page. We do not want to reload the page, we only want to close the carousel on backwards navigation.

Reply:
I have not been able to use `router.push` because it was not updating the history in anyway. I don't know the reason, but I believe it may be related to the path being identical to the last one, and I guess the router drops duplicate entries.

I didn't understand that the push had to be unique for the entire carousel. I will work on this as soon as possible.

Reply:
Adding a query param might work:

`router.push({ pathname: router.pathname, query: { carousel: true, ...router.query}}, router.asPath, { shallow: true }}`

Reply:
That's what I've been trying to do but it looks like the carousel is immediately closed after a 'router.push' or 'router.replace' call. This does not happen when calling native `window' APIs though.

Reply:
If you used `shallow: true` and it did that, then I'd guess something is rerendering that shouldn't.

Reply:
I believe this is due to line 73 of `modal.js`:
```
router.events.on('routeChangeStart', maybeOnClose)
```

The push call triggers this event, which then calls maybeOnClose. This makes sense as the native window API is not trigger router's events.